spec/models/assessment_detail_spec.rb

```diff
@@ -14,6 +14,7 @@
           :assessment_detail,
           :past_applications,
           entry: "",
+          additional_information: "qwerty",
```

Comment:
The issue with these tests is that they're in a loop which iterates through the categories. When the category is `past_applications` the setup blows up if `additional_information` isn't there. I guess I could have an `if category ==`, but this was quicker and I don't think it hurts..?

Likewise in the factory we were iterating through categories to create traits. In that case I've now updated it so that `additional_information` is only populated if the category is `past_applications`.

Reply:
Something I'm missing here... when I remove additional_information line ... and run everything in the spec 
 
`  bundle exec rspec spec/models/assessment_detail_spec.rb` 

 ...everything still passes. What am I missing?
```
          let!("#{category_type}1") do
            create(
              :assessment_detail,
              :"#{category_type}",
              created_at: 1.day.ago,
              planning_application: planning_application
            )
          end

          let!("#{category_type}2") do
            create(
              :assessment_detail,
              :"#{category_type}",
              created_at: Time.zone.now,
              planning_application: planning_application
            )
          end

          let!("#{category_type}3") do
            create(
              :assessment_detail,
              :"#{category_type}",
              created_at: 2.days.ago,
              planning_application: planning_application
            )
          end
```

Reply:
Ohhh...because then I fixed the issue in the test, and then fixed it in the factory, which renders fix one redundant 🤣 

Thanks for checking!

Reply:
I'll approve it now... 

BTW: this bit where you do a sort and then end the variables in numbers. Makes it difficult to reason - should 2 be first, should 1 be second?? and 3 last??

I would have used the suffixes  _now, _1_day_ago, _2_days_ago.     .... or some other scheme to indicate the variables age.

` eq([send("#{category_type}2"), send("#{category_type}1"), send("#{category_type}3")])
` 
 There's a rubocop that covers having numbers at the end of variables.

Reply:
So many things to refactor, so little time 😅